backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -16,21 +18,40 @@
 @Service
 public class ReleaseService implements ReleaseAmendingLawAndAllRelatedTargetLawsUseCase {
   private final LoadAmendingLawPort loadAmendingLawPort;
+  private final LoadAnnouncementPort loadAnnouncementPort;
   private final UpdateAmendingLawPort updateAmendingLawPort;
+  private final UpdateAnnouncementPort updateAnnouncementPort;
 
   public ReleaseService(
-      LoadAmendingLawPort loadAmendingLawPort, UpdateAmendingLawPort updateAmendingLawPort) {
+      LoadAmendingLawPort loadAmendingLawPort,
+      LoadAnnouncementPort loadAnnouncementPort,
+      UpdateAmendingLawPort updateAmendingLawPort,
+      UpdateAnnouncementPort updateAnnouncementPort) {
     this.loadAmendingLawPort = loadAmendingLawPort;
+    this.loadAnnouncementPort = loadAnnouncementPort;
     this.updateAmendingLawPort = updateAmendingLawPort;
+    this.updateAnnouncementPort = updateAnnouncementPort;
   }
 
   @Override
   public Optional<AmendingLaw> releaseAmendingLaw(Query query) {
-    final Optional<AmendingLaw> amendingLawOptional =
+    var amendingLawOptional =
```

Comment:
Mabe move this code closer to the `return` statement.
Then we have one block for the annoucements and one for the amendingLaws.